### ISSUE TITLE
[ENHANCEMENT] Implement RTTI

### ISSUE THREAD

**Issue by MasterEric** [larsiusprime/polymod#146]
https://haxe.org/manual/cr-rtti.html

By generating runtime type information for scripted classes, then parsing and accessing it as needed, significant numbers of `Reflect` calls will be avoided, providing major performance improvements on Native platforms.

Would have to test but probably doesn't provide relevant performance improvements on Web.

**Comment by TheRealJake12** [comment]
and this never happened... 😢

**Comment by PurSnake** [comment]
hmm

**Comment by EliteMasterEric** [comment]
Knowing a bit more, I don't know that it'd necessarily provide any savings over normal reflection.

If anyone can find a method of handling field accesses for scripted classes that's faster than standard Reflection (which in C++ literally just calls a switch/case block to the fields of the object), that would be great.

**Comment by EliteMasterEric** [comment]
yeah from my last investigation I don't know that any form of RTTI would actually be faster than reflection, I looked at how reflection gets done under the hood and it's already pretty fast. If there's a specific method of speeding up scripts that someone finds, please make a new issue or PR.
